### PR TITLE
Change dtype of sample_series and remove convert_series util

### LINKED ISSUE #710: Update sample_series fixture to have proper dtype
Currently we call `convert_series(sample_series, Categorical)`any time we want to use the `sample_series` fixture because its dtype is `object`. If we change the dtype to be `categorical` for pandas and dask and `string` for koalas, we can greatly minimize `convert_dtype` usage (probably won't be able to fully remove because there are some places where we pass different logical types to convert_series)

### PR DESCRIPTION
**Pull request by tamargrey**
- Changes sample_series to have the dtype that's valid for the `Categorical` logical type, which it gets inferred as. This means we can remove the `convert_series` util as there were only 9 more uses of it, and 8 of them just required a simple `astype` call
- Closes #710

### CODE REVIEW

File: woodwork/tests/accessor/test_accessor_utils.py [hunk 1]
```diff
@@ -21,14 +21,17 @@
 
 def test_init_series_valid_conversion_specified_ltype(sample_series):
     series = init_series(sample_series, logical_type='categorical')
-    assert series is not sample_series
+    assert series is sample_series
```
Comment (thehomebrewnerd) [inline woodwork/tests/accessor/test_accessor_utils.py:24]: I think with the change to the dtype of `sample_series` we now never test converting a series dtype with Koalas like we did before. Maybe we should update this to convert sample series to have a different dtype at the start in this case, and leave the rest of the test the same?

I think this is one test where having sample_series not have a `category` dtype was useful.

File: woodwork/tests/accessor/test_accessor_utils.py [hunk 2]
```diff
@@ -37,7 +40,7 @@ def test_init_series_valid_conversion_specified_ltype(sample_series):
 
 def test_init_series_valid_conversion_inferred_ltype(sample_series):
     series = init_series(sample_series)
-    assert series is not sample_series
+    assert series is sample_series
```
Comment (thehomebrewnerd) [inline woodwork/tests/accessor/test_accessor_utils.py:43]: Same as previous comment - I think we want to verify that the dtype gets changed properly by `init_series` in this test.

File: woodwork/tests/accessor/test_accessor_utils.py [hunk 3]
```diff
@@ -59,7 +62,7 @@ def test_init_series_all_parameters(sample_series):
                          metadata=metadata,
                          description=description,
                          use_standard_tags=False)
-    assert series is not sample_series
+    assert series is sample_series
```
Comment (thehomebrewnerd) [inline woodwork/tests/accessor/test_accessor_utils.py:65]: Same as above

**Review by thehomebrewnerd — COMMENTED** [review]
Looks pretty good, just a couple comments on the tests in `test_accessor_utils.py`

File: woodwork/tests/accessor/test_accessor_utils.py (hunk 1 above)
Comment (tamargrey) [reply woodwork/tests/accessor/test_accessor_utils.py:24]: ahh, got it. I can change the dtype back to `'object'` for this and the other init_series tests at the start to keep the test the same as it was before, if you think that makes sense

Comment (thehomebrewnerd) [reply woodwork/tests/accessor/test_accessor_utils.py:24]: Yeah, I think that makes sense.

Comment (tamargrey) [reply woodwork/tests/accessor/test_accessor_utils.py]: okay, changed!

**Review by thehomebrewnerd — APPROVED** [review]
LGTM!

**Pull request merged**